src/RageSound.cpp

```diff
@@ -499,33 +494,25 @@ int RageSound::GetSourceFrameFromHardwareFrame( std::int64_t iHardwareFrame, boo
  */
 float RageSound::GetPositionSeconds( bool *bApproximate, RageTimer *pTimestamp ) const
 {
-	/* Get our current hardware position. */
-	std::int64_t iCurrentHardwareFrame = SOUNDMAN->GetPosition( pTimestamp );
-
-	/* Lock the mutex after calling SOUNDMAN->GetPosition().  We must not make driver
-	 * calls with our mutex locked (driver mutex < sound mutex). */
-	LockMut( m_Mutex );
-
-	if( bApproximate )
-		*bApproximate = false;
+	// Get our current hardware position.
+	std::int64_t iCurrentHardwareFrame = SOUNDMAN->GetPosition(pTimestamp);
+	float fSampleRate = static_cast<float>(m_pSource->GetSampleRate());
+	// Lock the mutex after calling SOUNDMAN->GetPosition().
+	LockMut(m_Mutex);
 
 	/* If we're not playing, just report the static position. */
-	float sampleRate = static_cast<float>(samplerate());
 	if( !IsPlaying() )
-		return m_iStoppedSourceFrame / sampleRate;
+		return static_cast<float>(m_iStoppedSourceFrame) / fSampleRate;
```

Comment:
The static casts are implicit (here and below). If you want to be explicit (which is fine), then it would make sense to also update the one on line 515.

Reply:
I'll make it so the one on line 515 matches this style 👍